### PR TITLE
Basic service implementation and ocis-web extension

### PR DESCRIPTION
**Pull request by kulmann**
Once the service is running, the following request can be used to provide a 2 ways testable file:

create a settings bundle:

```console
micro call com.owncloud.web.ocis-settings BundleService.CreateSettingsBundle '{
    "settingsBundle": {
        "extension": "Phoenix",
        "key": "theming",
        "settings": [
            {
                "key": "dark",
                "displayName": "Dark Theme",
                "stringValue": {
                    "default": "Dark",
                    "required": false,
                    "placeholder": "Select a theme"
                }
            },
            {
                "key": "light",
                "displayName": "Light Theme",
                "stringValue": {
                    "default": "Light",
                    "required": false,
                    "placeholder": "Select a theme"
                }
            }
        ]
    }
}'
```

read the settings bundle:

```console
micro call com.owncloud.web.ocis-settings BundleService.GetSettingsBundle '{"extension": "Phoenix", "key": "theming"}'
```

The output of that last command should be:

```json
{
    "settingsBundle": {
        "extension": "Phoenix",
        "key": "theming",
        "settings": [
            {
                "key": "dark",
                "displayName": "Dark Theme",
                "stringValue": {
                    "default": "Dark",
                    "placeholder": "Select a theme"
                }
            },
            {
                "key": "light",
                "displayName": "Light Theme",
                "stringValue": {
                    "default": "Light",
                    "placeholder": "Select a theme"
                }
            }
        ]
    }
}
```

### CODE REVIEW

File: pkg/store/filesystem/store.go [hunk 1]
```diff
@@ -106,18 +107,30 @@ func (s Store) Write(rec *proto.SettingsBundle) (*proto.SettingsBundle, error) {
 		return nil, fmt.Errorf(emptyKeyError)
 	}
 
-	contents, err := json.Marshal(rec)
-	if err != nil {
-		s.Logger.Err(err).Msg("record could not be marshalled")
+	marshaler := jsonpb.Marshaler{}
+	recordPath := path.Join(s.mountPath, buildFileNameFromBundle(rec))
+	if err := ioutil.WriteFile(recordPath, []byte{}, 0644); err != nil {
```
Comment (refs) [inline pkg/store/filesystem/store.go:112]: this is a patch. I'm pretty sure there is a way to create a file without needing to write something into it

File: pkg/proto/v0/settings.proto [hunk 2]
```diff
@@ -90,4 +90,5 @@ message ListOption {
     int64 intValue = 2;
   }
   bool selected = 3;
+  string displayValue = 4;
```
Comment (refs) [inline pkg/proto/v0/settings.proto:93]: Let's stick to the [language style guide](https://developers.google.com/protocol-buffers/docs/style#message-and-field-names) and use `underscode_separated_names`. I would also have to update it on the accounts service.

File: pkg/store/filesystem/io.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package store
```
Comment (refs) [inline pkg/store/filesystem/io.go:1]: could you rename `io.go` to `main.go`? I would do the same on the occis-accounts service, for consistency.

Comment (kulmann) [reply pkg/store/filesystem/io.go:1]: But its not main, it is only the io, i.e. marshaling and unmarshaling with proper file locks.

File: pkg/proto/v0/settings.proto (hunk 2 above)
Comment (kulmann) [reply pkg/proto/v0/settings.proto:93]: done :-)

Comment (kulmann) [reply pkg/proto/v0/settings.proto:93]: but in the other PR. will merge it into this one eventually

**Pull request merged**